frontend/src/utils/playback-profiles/helpers/codec-profiles.ts

```diff
@@ -26,7 +26,17 @@ import {
 function getGlobalMaxVideoBitrate(): number | undefined {
   let isTizenFhd = false;
 
-  if (isTizen() && window.webapis) {
+  if (
+    isTizen() &&
+    'webapis' in window &&
+    typeof window.webapis === 'object' &&
+    window.webapis &&
+    'productinfo' in window.webapis &&
+    typeof window.webapis.productinfo === 'object' &&
+    window.webapis.productinfo &&
+    'isUdPanelSupported' in window.webapis.productinfo &&
+    typeof window.webapis.productinfo.isUdPanelSupported === 'function'
+  ) {
```

Comment:
FYI, our idea is to move completely out from our custom profile detections and use MediaCapabilities API, so this file is indeed deprecated.

You can coordinate with @ThibaultNocchi, because it will be a huge step forward better playback and this is the major legacy thing we have in this client as of today.

Reply:
Do you want to coordinate removing it in this PR or for future work?

Reply:
@aweebs Future work, but I would rule modifying this file altogether in this PR, it's indeed introducing more type errors (even after applying the Vuetify 3.1.11 update to this branch).

In my discussions with @ThibaultNocchi, we thought the best approach for MediaCapabilities would be to phase out the "generate a device profile of all codecs at startup" approach and move to a "test for compatibility before starting the playback", based on the codec information of the selected media streams given by the server. We should test for video and audio independently and report as compatible all the web-based subtitlte formats (WebVTT) + ASS (thanks to JASSUB).

If a file is not compatible, we should report to server that we're compatible with H264 and AAC, which is a good default combo that all browsers supports. There's no point in checking for MP3, H263, etc... Maybe we can also test for H265 as Edge on Windows and Safari supports it.

That should be done exclusively in a new PR, as removing all this logic will break playback.

Reply:
The added type errors are because of an outdated version of typescript. I realized I had been using 5.x.x since it was announced recently. Even upgrading to a later version of 4.x.x resolves these. I'll add the typescript update to this PR, it doesn't seem to impact anything other than our typing.

I would like to get rid of this type error though, the other cheating way to get rid of this is with an any cast
```
  if (isTizen()) {
    isTizenFhd = !(window as any).webapis.productinfo.isUdPanelSupported();
  }
```

I'd prefer to keep the above now that it won't cause additional type errors. Even if it's very verbose, we're pretty close to getting of all of them :D